test/Parse/switch.swift

```diff
@@ -502,3 +502,25 @@ switch x {
 case _:
   @unknown let _ = 1 // expected-error {{unknown attribute 'unknown'}}
 }
+
+switch Whatever.Thing {
+case .Thing:
+  break
+@unknown(garbage) case _: // expected-error {{unexpected '(' in attribute 'unknown'}}
+  break
+}
+switch Whatever.Thing {
+case .Thing:
+  break
+@unknown // expected-note {{attribute already specified here}}
+@unknown // expected-error {{duplicate attribute}}
+case _:
+  break
+}
+switch Whatever.Thing {
+case .Thing:
+  break
+@unknown @garbage(foobar) // expected-error {{unknown attribute 'garbage'}}
+case _:
```

Comment:
Ah, if you want to check that the `@unknown` isn't getting dropped, you'll have to leave out the `.Thing` case and check that there's still a warning for it.

Reply:
I see. Thanks!